### DISCUSSION TITLE
about csv&xml based annotations

### DISCUSSION THREAD

**Discussion by nirolada** [Nelson-Gon/cytounet#32] [Ideas]
I suggested a class of annotations reader for reading and validating annotation files in CSV & XML formats.
What do you think about it? Should I start working on it? Any other suggestions?

**Comment by Nelson-Gon** [comment]
I think it sounds good. I would only add that we need to figure out how to use these in training but I imagine you've thought about this already. Thanks!

**Reply by nirolada** [reply]
What do you prefer?
1. Read the data and create a python dictionary for each row.
2. Read the data and create a pandas (a python package like Numpy) dataframe for the whole data.

**Reply by Nelson-Gon** [reply]
I think a `DataFrame` would be more resource friendly and possibly easier to work with. That said, I have not yet had time to take a look at the best way to feed this into the Keras training pipeline.

**Comment by nirolada** [comment]
This is a good point. Can you check and fill me in what type input of
annotations does keras accept?

**Reply by Nelson-Gon** [reply]
I will look into it more and will respond by Tuesday.

**Reply by Nelson-Gon** [reply]
@nirolada [This](https://github.com/keras-team/keras/issues/10196)  is the closest I have found. From the linked StackOverflow question, it seems the best bet is to write a custom data generator.  In response to the kind of data, it seems there is no native support for labels with `Keras`.
